### PR TITLE
[YANG] Update device_metadata to add suppress-fib-pending field

### PR DESCRIPTION
**Pull request by Gfrom2016**
#### Why I did it
 BGP pending FIB suppression is introduced in #12853, hence updating the YANG model.

##### Work item tracking
- Microsoft ADO **(number only)**: 28398135

#### How I did it
Added a new field 'suppress-fib-pending' in the DEVICE_METADATA YANG model table.
Added UT for YANG model changes.

#### How to verify it
Run PR tests.
Apply empty.json file to verify:
```
admin@str2-7050cx3-acs-02:~$ sudo config apply-patch empty.json 
Patch Applier: Patch application starting.
Patch Applier: Patch: []
Patch Applier: Getting current config db.
Patch Applier: Simulating the target full config after applying the patch.
Patch Applier: Validating all JsonPatch operations are permitted on the specified fields
Patch Applier: Validating target config does not have empty tables, since they do not show up in ConfigDb.
Patch Applier: Sorting patch updates.
Patch Applier: The patch was converted into 0 changes.
Patch Applier: Applying 0 changes in order.
Patch Applier: Verifying patch updates are reflected on ConfigDB.
Patch Applier: Patch application completed.
Patch applied successfully.
admin@str2-7050cx3-acs-02:~$ 
```

Signed-off-by: zitingguo-ms <zitingguo@microsoft.com>

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by lolyu — APPROVED** [review]
LGTM

**Review by yutongzhang-microsoft — APPROVED** [review]
LGTM

**Comment by Gfrom2016** [comment]
@wen587 Could you help to review this?

File: src/sonic-yang-models/yang-models/sonic-device_metadata.yang [hunk 1]
```diff
@@ -230,6 +230,15 @@ module sonic-device_metadata {
                                  otherwise the maximum available buffers (which are read from SAI) will be
                                  created, regardless of the CONFIG_DB buffers configuration.";
                 }
+
+                leaf suppress-fib-pending {
+                    type stypes:admin_mode;
+                    default enabled;
```
Comment (wen587) [inline src/sonic-yang-models/yang-models/sonic-device_metadata.yang:236]: I saw the master PR default is `disabled`. Why we change to enabled here?

Comment (Gfrom2016) [reply src/sonic-yang-models/yang-models/sonic-device_metadata.yang]: changed to 'disabled'

**Review by wen587 — APPROVED** [review]

**Review by StormLiangMS — APPROVED** [review]
LGTM

**Pull request merged**